zebra-consensus/src/chain.rs

```diff
@@ -137,52 +141,40 @@ where
         };
 
         async move {
-            // TODO(teor): in the post-sapling checkpoint range, allow callers
-            //             to use BlockVerifier, CheckpointVerifier, or both.
-
             // Call a verifier based on the block height and checkpoints.
             if is_higher_than_max_checkpoint(height, max_checkpoint_height) {
-                // Log a message on early high blocks.
+                // Log a message on unexpected out of order blocks.
+                //
                 // The sync service rejects most of these blocks, but we
                 // still want to know if a large number get through.
-                //
-                // This message can also happen if we keep getting unexpected
-                // low blocks. (We can't distinguish between these cases, until
-                // we've verified the blocks.)
-                if is_unexpected_high_block {
-                    tracing::debug!(?height, "unexpected high block, or recent unexpected low blocks");
+                if is_unexpected_gap {
+                    tracing::debug!("large block height gap: this block or the previous block is out of order");
                 }
 
-                block_verifier
+                let verified_hash = block_verifier
                     .ready_and()
                     .await?
                     .call(block.clone())
                     .await?;
+                assert_eq!(verified_hash, hash, "block verifier returned wrong hash: hashes must be equal");
             } else {
-                checkpoint_verifier
-                    .expect("Missing checkpoint verifier: verifier must be Some if max checkpoint height is Some")
+                                let verified_hash = checkpoint_verifier
```

Comment:
did something go wrong with rustfmt here?

Reply:
Weird that it passed my rustfmt, and also our rustfmt checks.

Reply:
Fixed and squashed into f9088027f207577b08b0a7450d1ce05bc2e31c54.